File: src/roar_net_api/api/operations/objective_value_increment.py
# SPDX-FileCopyrightText: © 2025 Authors of the roar-net-api-py project <https://github.com/roar-net/roar-net-api-py/blob/main/AUTHORS>
#
# SPDX-License-Identifier: Apache-2.0

from typing import Optional, Protocol, TypeVar, Union

Solution = TypeVar("Solution", contravariant=True)


class SupportsObjectiveValueIncrement(Protocol[Solution]):
    def objective_value_increment(self, solution: Solution) -> Optional[Union[int, float]]: ...

File: julia/src/operations/objective_value_increment.jl

# SPDX-FileCopyrightText: © 2025 Authors of the roar-net-api-py project <https://github.com/roar-net/roar-net-api-py/blob/main/AUTHORS>
#
# SPDX-License-Identifier: Apache-2.0
"""
    objective_value_increment(move::Move, solution::Solution) -> value::Union{Nothing, <:Real}

Return the objective value increment for applying `move` to `solution`,
or `nothing` if the move is not applicable. Used by local search algorithms.
"""
function objective_value_increment(::Move, ::Solution)::Union{Nothing, <:Real} end
